projects/plugins/boost/app/modules/cache/Boost_Cache_Settings.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+/*
+ * Cache settings class.
+ * Settings are stored in a file in the boost-cache directory.
+ * This file is shared by autoloaded code and code loaded by advanced-cache.php
+ */
+
+class Boost_Cache_Settings {
+	private $settings   = array();
+	private $last_error = '';
+	private $config_file_path;
+	private $config_file;
+
+	public function __construct() {
+		$this->config_file_path = WP_CONTENT_DIR . '/boost-cache/';
+		$this->config_file      = $this->config_file_path . 'config.php';
+		$this->init_settings();
+	}
+
+	/*
+	 * Load the settings from the config file, if available.
+	 */
+	private function init_settings() {
+
+		if ( ! file_exists( $this->config_file_path ) ) {
+			mkdir( $this->config_file_path, 0755, true ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_mkdir
+		}
+
+		if ( ! file_exists( $this->config_file ) ) {
+			$result = $this->set( array( 'enabled' => false ) );
+			if ( ! $result ) {
+				return false;
+			}
```

Comment:
At this point, if `$result` is true, don't we already know the value of settings? Is there a reason to read it back?

Reply:
$result is true only if the setting was successfully written to the config file. We're not doing anything with that return value yet anyway, so I'll remove it. It would be better off handled in the set() function, where it will log it for an admin_notice perhaps to inform the user.

Reply:
Wait, if you remove that check then it will try to read the config file after it fails to save it and overwrite the last_error message about the save failing.

My point was that if $result is true, you can just return the value you wrote. :) But I guess that's only a tiny optimization for the smallest edge-case.  And the last_error thing doesn't super matter until we implement error logging.